### PR TITLE
Update Gradle build scripts

### PR DESCRIPTION
**Pull request by PatrykMis**

### CODE REVIEW

**Comment by lberrymage** [comment]
Hi! Can you explain what exactly this is doing? Is this supposed to be a more modern way of setting up the build scripts or something?

**Comment by PatrykMis** [comment]
Exactly, it is explained at <https://developer.android.com/build/migrate-to-kotlin-dsl#migrate-buildscript>

File: build.gradle.kts [hunk 1]
```diff
@@ -1,17 +1,9 @@
 plugins {
+    id("com.android.application") version "8.0.1" apply false
+    id("com.android.library") version "8.0.1" apply false
```
Comment (lberrymage) [inline build.gradle.kts]: Is this line necessary for an application?

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -3092,9 +3107,14 @@
...
       </component>
+      <component group="org.jetbrains.kotlin.android" name="org.jetbrains.kotlin.android.gradle.plugin" version="1.8.20">
+         <artifact name="org.jetbrains.kotlin.android.gradle.plugin-1.8.20.pom">
+            <sha256 value="a805594a79703ae9fdaafbbf42c6cf00d57355213ce728b4c6716397ff949178" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.jetbrains.kotlin.plugin.serialization" name="org.jetbrains.kotlin.plugin.serialization.gradle.plugin" version="1.8.20">
          <artifact name="org.jetbrains.kotlin.plugin.serialization.gradle.plugin-1.8.20.pom">
-            <sha256 value="51e504aa44c7c572dfbc1ee08d7b447f22374511efbdbd8fabb46371187f1a1c" origin="Generated by Gradle"/>
+            <sha256 value="b81d0f71998ac7ad92315773746ee702d402e796432e0d360789d8f59923c762" origin="Generated by Gradle"/>
```
Comment (lberrymage) [inline gradle/verification-metadata.xml]: Why has this hash changed?

**Review by lberrymage — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (PatrykMis) [reply build.gradle.kts]: It seams not, even if whole Android Developer documentation suggests it. Fixed.

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (PatrykMis) [reply gradle/verification-metadata.xml]: Really don't know why, so I've researched this, tried on different OSs and different machines. It seams if something in the code / dependencies is changed, the hash of this particular artifact is changing too while generating by Gradle. I've rebased pull request and this one from latest rebase / commit is reproduceable on all platforms.

File: gradle/verification-metadata.xml [hunk 3]
```diff
@@ -3092,9 +3102,14 @@
...
       </component>
+      <component group="org.jetbrains.kotlin.android" name="org.jetbrains.kotlin.android.gradle.plugin" version="1.8.21">
+         <artifact name="org.jetbrains.kotlin.android.gradle.plugin-1.8.21.pom">
+            <sha256 value="85e4a6a46ce6d234d12421ef3865664714e8afbf7287a651aa580646821340a6" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.jetbrains.kotlin.plugin.serialization" name="org.jetbrains.kotlin.plugin.serialization.gradle.plugin" version="1.8.21">
          <artifact name="org.jetbrains.kotlin.plugin.serialization.gradle.plugin-1.8.21.pom">
-            <sha256 value="78403d230b0bcf54f94593dbd18267d2f4a275dc3ea44d2ec2cade26537a4baa" origin="Generated by Gradle"/>
+            <sha256 value="8426766c4866369ca9ac2c9ad94879059a34b72e0b6a8e20317fdb55f967fc26" origin="Generated by Gradle"/>
```
Comment (lberrymage) [inline gradle/verification-metadata.xml:3112]: I'm not sure why Gradle generates this for you, but this hash doesn't verify for me.

**Review by lberrymage — CHANGES_REQUESTED** [review]
I'm not sure what causes the kotlinx.serialization hash to be different for you, but everything else looks good. Thank you!

**Comment by lberrymage** [comment]
Closing as implemented in https://github.com/accrescent/accrescent/commit/383488db9eeaa1d1042fa08ef17691b7d29eacae

**Pull request closed without merging**